src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,169 @@
+            if (!(parameters.get(OVERLAP_RATE_FIELD) instanceof Number)) {
+                throw new IllegalArgumentException(
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] cannot be cast to [" + Number.class.getName() + "]"
+                );
+            }
+            BigDecimal overlap_rate = new BigDecimal(String.valueOf(parameters.get(OVERLAP_RATE_FIELD)));
+            if (overlap_rate.compareTo(BigDecimal.ZERO) < 0 || overlap_rate.compareTo(OVERLAP_RATE_UPPER_BOUND) > 0) {
+                throw new IllegalArgumentException(
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] must be between 0 and " + OVERLAP_RATE_UPPER_BOUND
+                );
+            }
+        }
+
+        if (parameters.containsKey(TOKENIZER_FIELD) && !(parameters.get(TOKENIZER_FIELD) instanceof String)) {
+            throw new IllegalArgumentException(
+                "fixed length parameter [" + TOKENIZER_FIELD + "] cannot be cast to [" + String.class.getName() + "]"
+            );
+        }
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. tokenizer the analyzer tokenizer in opensearch, please check https://opensearch.org/docs/latest/analyzers/tokenizers/index/
+     * 2. token_limit the token limit for each chunked passage
+     * 3. overlap_rate the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * 4. max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(String content, Map<String, Object> parameters) {
+        // prior to chunking, parameters have been validated
+        int tokenLimit = DEFAULT_TOKEN_LIMIT;
+        BigDecimal overlap_rate = new BigDecimal(String.valueOf(DEFAULT_OVERLAP_RATE));
+        int maxTokenCount = DEFAULT_MAX_TOKEN_COUNT;
+
+        String tokenizer = DEFAULT_TOKENIZER;
+
+        if (parameters.containsKey(TOKEN_LIMIT_FIELD)) {
+            tokenLimit = ((Number) parameters.get(TOKEN_LIMIT_FIELD)).intValue();
+        }
+        if (parameters.containsKey(OVERLAP_RATE_FIELD)) {
+            overlap_rate = new BigDecimal(String.valueOf(parameters.get(OVERLAP_RATE_FIELD)));
+        }
+        if (parameters.containsKey(MAX_TOKEN_COUNT_FIELD)) {
+            maxTokenCount = ((Number) parameters.get(MAX_TOKEN_COUNT_FIELD)).intValue();
+        }
+        if (parameters.containsKey(TOKENIZER_FIELD)) {
+            tokenizer = (String) parameters.get(TOKENIZER_FIELD);
+        }
+
+        List<String> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> passages = new ArrayList<>();
+
+        String passage;
+        int startToken = 0;
+        BigDecimal overlapTokenNumberBigDecimal = overlap_rate.multiply(new BigDecimal(String.valueOf(tokenLimit)))
```

Comment:
Since it's not using very precise value of the overlap rate, do we really need to use BigDecimal?

Reply:
Maybe precise value for overlap rate is related to parameter validation? Please correct me if I am wrong @chishui

Reply:
According to our today's discussion, we will use double instead of big decimal. I will resolve this conversation.